OpenRA.Mods.Common/Traits/HitShape.cs

```diff
@@ -73,6 +73,8 @@ protected override void Created(Actor self)
 			base.Created(self);
 		}
 
+		bool ITargetablePositions.AlwaysEnabled { get { return Info.EnabledByDefault; } }
```

Comment:
`EnabledByDefault` doesn't necessarily mean *always* enabled - surly?

Reply:
At first I was wondering about that, too, but the base `ConditionalTrait` currently sets 
`EnabledByDefault = RequiresCondition == null || RequiresCondition.Evaluate(VariableExpression.NoVariables);`

and `HitShape` doesn't override that. So at least in this context it should be correct.

Reply:
Please don't do this! `EnabledByDefault` is a horrible hack around the code not yet supporting arbitrary starting conditions (via `ConditionInit`).  We really don't want to make our low level abstractions depend on this broken code.

Reply:
I see. Would it be ok to check
> RequiresCondition == null || RequiresCondition.Evaluate(VariableExpression.NoVariables)

here directly?

Reply:
The null check is fine, but the `.Evaluate` isn't.  The null check by itself should satisfy your goal here.